base_layer/wallet/src/base_node_service/monitor.rs

```diff
@@ -95,27 +106,38 @@ impl<T: WalletBackend + 'static> BaseNodeMonitor<T> {
 
     async fn monitor_node(&mut self) -> Result<(), BaseNodeMonitorError> {
         loop {
+            let start = Instant::now();
             let mut client = self
                 .wallet_connectivity
                 .obtain_base_node_wallet_rpc_client()
                 .await
                 .ok_or(BaseNodeMonitorError::NodeShuttingDown)?;
+            trace!(
+                target: LOG_TARGET,
+                "Obtain RPC client {} ms",
+                start.elapsed().as_millis()
+            );
 
             let base_node_id = match self.wallet_connectivity.get_current_base_node_id() {
                 Some(n) => n,
                 None => continue,
             };
 
+            let start = Instant::now();
             let tip_info = client.get_tip_info().await?;
-
             let chain_metadata = tip_info
                 .metadata
                 .ok_or_else(|| BaseNodeMonitorError::InvalidBaseNodeResponse("Tip info no metadata".to_string()))
                 .and_then(|metadata| {
                     ChainMetadata::try_from(metadata).map_err(BaseNodeMonitorError::InvalidBaseNodeResponse)
                 })?;
+            // Note: Do not use an RPC ping call to measure latency here, as the time measured for that always follows
+            // the time it takes for the metadata call, which is evident as well when the base node gets busy and will
+            // often take longer than the metadata call itself. Having two longish RPC calls with no added value
+            // results in more RPC request timeouts than is necessary.
+            // TODO: Find root cause of RPC ping calls taking approximately as long as RPC fetch meta data calls
```

Comment:
This seems more like a comment for a PR than a comment that is helpful for future developers. I would suggest removing it